src/temgymbasic/components.py

```diff
@@ -146,6 +155,114 @@ def gui_wrapper():
         return SampleGUI
 
 
+class PotentialSample(Sample):
+    def __init__(
+        self,
+        z: float,
+        potential: NDArray,
+        pixel_size: float,
+        shape: Tuple[int, int],
+        rotation: Radians = 0.,
+        flip_y: bool = False,
+        center: Tuple[float, float] = (0., 0.),
+        name: Optional[str] = None,
+    ):
+        super().__init__(name=name, z=z)
+        self.phi = potential
+        self.pixel_size = pixel_size
+        self.shape = shape
+        self.rotation = rotation  # degrees
+        self.flip_y = flip_y
+        self.center = center
+        self.dphi_dy, self.dphi_dx = np.gradient(potential)
+
+    def set_center_px(self, center_px: Tuple[int, int]):
```

Comment:
I am not sure this function is used ? (it's copy/paste from `Detector` and I can't see how it would be used...)

Reply:
I am working on it now, almost have a new change ready to go

Reply:
That uses interpolation and is a lot simpler.